### ISSUE TITLE
Everything is fine... But Notification doesn't show up.

### ISSUE THREAD

**Issue by Taremeh** [EddyVerbruggen/nativescript-local-notifications#40]
I installed the plugin for Nativescript 2.4 (Android).
So far I used the demo code and everything seemed fine.

I also got the console.log, that the Notification is scheduled.
But there is no Notification showing up... And I don't get any errors...

Someone experiencing the same problem? Is there a solution?

```
  LocalNotifications.schedule([{
    id: 1,
    title: 'The title',
    body: 'Recurs every minute until cancelled',
  }]).then(
      function() {
        console.log("Notification scheduled");
      },
      function(error) {
        console.log("scheduling error: " + error);
      }
  )
```

**Comment by Taremeh** [comment]
Same issue for {NS} 2.5. Upgraded everything today and still no clue what's wrong...
If I check currently scheduled Notifications they don't display:

```
JS: Notification scheduled (<--- This should mean: success, everything is fine)
JS: ID's: 0 (<--- But nothing here :( )
```

**Comment by EddyVerbruggen** [comment]
You'll need to pass in the `at` property, otherwise the notification is triggered immediately and you won't see it as your app is already in the foreground.

**Comment by Taremeh** [comment]
@EddyVerbruggen Unfortunately that doesn't help:

```
LocalNotifications.schedule([{
        id: 61,
        title: 'The title',
        body: 'Recurs every minute until cancelled',
        at: new Date(new Date().getTime() + (5 * 1000)) // 5 seconds from now
    }]).then(
        function() {
            console.log("Notification scheduled");
        },
        function(error) {
            console.log("scheduling error: " + error);
        }
    );
```

Console says:
```
JS: Notification scheduled
JS: ID's: 0,1,61,59,60

```

**Comment by EddyVerbruggen** [comment]
Can you please mail me your app so I can check it out?

**Comment by Taremeh** [comment]
It's actually on Github. I created a branch.

https://github.com/Taremeh/apphovenAlpha/tree/notification-plugin-issue

Here the notification code:
https://github.com/Taremeh/apphovenAlpha/blob/notification-plugin-issue/apphovenAlpha/app/pages/home/home.component.ts#L38-L58

**Comment by EddyVerbruggen** [comment]
You branch doesn't compile because of [this line](https://github.com/Taremeh/apphovenAlpha/blob/master/apphovenAlpha/app/app.routing.ts#L7):

<img width="983" alt="screen shot 2017-02-11 at 20 41 58" src="https://cloud.githubusercontent.com/assets/1426370/22856863/aa47b148-f09a-11e6-8a95-99fbb8d7fe3d.png">

Once that works, will it be easy and obvious to reproduce the issue?

And just checking: you did close your app within 5 seconds after scheduling that notification, right?

**Comment by Taremeh** [comment]
Unfortunately not... I deleted the composer (and even search component).
Still the same issue. 

Btw. Would it be possible to create a notification while the app is running?

**Comment by EddyVerbruggen** [comment]
I'm not sure what your answers refer to. Did you fix the branch so I can try and reproduce the issue?

And did you close the app within 5 sec after scheduling the notification?

**Comment by cagatayoncul** [comment]
Same problem on android i am using 2.4.3

**Comment by cagatayoncul** [comment]
Ok i tested on real device and its working. On genymotion emulator notifications not working.

**Comment by francorobles** [comment]
Hi @EddyVerbruggen, I am planning to use this. Is this already installed?

**Comment by usmanakram1996** [comment]
@Taremeh  please help same issue notifications get scheduled but don't show up!

**Comment by Taremeh** [comment]
It's still a big pain. I could not get it to work properly... Unfortunately there is (still) no documentation about how to use & build native notifications with Nativescript... But maybe @EddyVerbruggen could provide some help? 😄 
I'd still love to know how to create a notification, which persists until an app event drops it. (Not a scheduled one, but one that pops up while using the app).

**Comment by Danziger** [comment]
@Taremeh @usmanakram1996 @francorobles I managed to get it working on my app by setting `forceShowWhenInForeground: true`, but I had to do some changes in [`local-notifications.ios.ts:182`](https://github.com/Danziger/nativescript-local-notifications/blob/master/src/local-notifications.ios.ts#L182).

I have opened a [PR with those changes](https://github.com/EddyVerbruggen/nativescript-local-notifications/pull/103), but it might take a while until we get them, as it looks like @EddyVerbruggen is now working on a major refactor on the Android part.

Tested it on real devices, not sure it would work in an iOS emulator.
